terraform/deployments/govuk-publishing-infrastructure/frontend_memcache.tf

```diff
@@ -0,0 +1,43 @@
+locals {
+  frontend_memcached_name = "frontend-memcached-${local.cluster_name}"
+}
+
+resource "aws_elasticache_subnet_group" "frontend_memcached" {
+  name       = "frontend-memcached"
+  subnet_ids = local.elasticache_subnets
+}
+
+resource "aws_security_group" "frontend_memcached" {
+  name        = local.frontend_memcached_name
+  vpc_id      = local.vpc_id
+  description = "${local.frontend_memcached_name} memcached instance"
+  tags = {
+    Name = local.frontend_memcached_name
+  }
+}
+
+resource "aws_elasticache_cluster" "frontend_memcached" {
+  cluster_id = "frontend-memcached-${local.cluster_name}"
+
+  engine          = "memcached"
+  engine_version  = "1.6.6"
+  node_type       = var.frontend_memcached_node_type
+  num_cache_nodes = 1 # TODO: consider whether failover is needed
+
+  apply_immediately    = var.govuk_environment != "production"
+  parameter_group_name = "default.memcached1.6"
+  subnet_group_name    = aws_elasticache_subnet_group.frontend_memcached.name
+  security_group_ids   = [aws_security_group.frontend_memcached.id]
+  tags = {
```

Comment:
do we want to add the default tags too? I saw that you removed it in the Redis ones too so might be expected here.

Reply:
Ah so there's a pretty recent [new feature](https://www.hashicorp.com/blog/default-tags-in-the-terraform-aws-provider) where it now does that for us (see [main.tf line 28](https://github.com/alphagov/govuk-infrastructure/blob/176ae02/terraform/deployments/govuk-publishing-infrastructure/main.tf#L28))

Reply:
I could have inlined `local.default_tags` into `provider.aws.default_tags` but I left it because there are still some resource types where the provider defaults don't get applied (because ugly Amazon API reasons) 🤷